src/utils.es

```diff
@@ -0,0 +1,30 @@
+import getDaysInMonth from 'date-fns/getDaysInMonth';
+import formatDate from 'date-fns/format';
+
+export const pad = (number, digits = 2) => String(number).padStart(digits, 0);
+export const arrayOf = (number) => Array.from(new Array(number)).map((_, index) => index);
+
+export const toDate = (isoDate) => {
+  const [year, month, day] = isoDate.split('-').map((item) => parseInt(item, 10));
+  return new Date(year, month - 1, day);
+};
+
+export const getMonth = (date) => new Date(date.getFullYear(), date.getMonth());
+
+export const getISOMonth = (current) => (
+  formatDate(current, 'yyyy-MM')
+);
+
+export const getISODate = (date) => formatDate(date, 'yyyy-MM-dd');
```

Comment:
you didn't really get rid of ISOs, instead you switch incoming props to Date but use ISOs for comparison, which is basically alphabetical check. It relies on per-character check of every element in a string eg. 'abc' > 'abd' and depends on the browser implementation to compare. To have this implemented properly you need to work with date objects/numbers (use numbers for comparison)